lxd/storage/backend_lxd.go

```diff
@@ -8048,7 +8048,14 @@ func (b *lxdBackend) ImportInstance(inst instance.Instance, poolVol *backupConfi
 	volStorageName := project.Instance(inst.Project().Name, inst.Name())
 
 	// Copy the volume's config so VolumeDBCreate can safely modify the copy if needed.
-	vol := b.GetNewVolume(volType, contentType, volStorageName, volumeConfig)
+	// Don't generate a new UUID as we are importing an existing instance's volume.
+	vol := b.GetVolume(volType, contentType, volStorageName, volumeConfig)
+
+	// In case the instance got lost on an old LXD which didn't set a volume's UUID,
+	// ensure we set a new one.
+	if vol.Config()["volatile.uuid"] == "" {
```

Comment:
just a nit, but we could avoid calling vol.Config() twice if its empty by storing it in a temporary variable.